packages/abi/src/matchers/sway-type-matchers.ts

```diff
@@ -104,9 +110,8 @@ export function createMatcher<T>(mappings: Record<SwayType, T>) {
 
     for (const [key, matcher] of swayTypeMatcherEntries) {
       if (matcher(swayType)) {
-        const mapping = mappings[key as SwayType];
-        if (mapping) {
-          return mapping;
+        if (key in mappings) {
+          return mappings[key as SwayType];
         }
```

Comment:
Let's add a test for this functionality.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3247/commits/ead2f0316ba36a947d9a75bccfa7477b57cc5540